surfinBH/_fit_evaluators/fit_7dq4Emri.py

```diff
@@ -0,0 +1,267 @@
+            (https://doi.org/10.1086/151796)
+            """
+ 
+            Z1 = 1+(1-chiz**2)**(1/3)*((1+chiz)**(1/3)+(1-chiz)**(1/3))
+            Z2 = (3*chiz**2+Z1**2)**(1/2)
+            r_ISCO = 3+Z2-np.sign(chiz)*((3-Z1)*(3+Z1+2*Z2))**(1/2)
+    
+            return r_ISCO
+
+        def eval_emriChif(q, chiA):
+            """
+            Estimate remnant spin for EMRI binary following Eq. (7) 
+            presented in THE PAPER.    
+            """
+            r_ISCO = eval_r_isco(chiA[2])
+            E_ISCO = np.sqrt(1-2/(3*r_ISCO))
+            L_ISCO = np.array([0,0,2/(3*3**(1/2))*(1+2*np.sqrt(3*r_ISCO-2))])
+                      
+            return chiA + 1/q*(L_ISCO-2*chiA*E_ISCO)  
+
+        def extrap_chif(x, fit_key):
+            """
+            The model evaluates remnant spin using GPR fit up to the upper bound
+            of the EMRI training region (q = 1000). At q > 1000, it returns the 
+            expected EMRI limit (Eq. (7) in THE PAPER). There is a transition 
+            function (Eq. (9) in THE PAPER) connecting smoothly these 
+            two regions. The transition takes place from q=1000 to q=2000.
+            """
+            x_i = x.copy()
+            x_i[0] = self.qEmriMax
+            
+            """This if statement defines the width of the transition region"""
+            if x[0] < 2*self.qEmriMax:
+                tmp = self._evaluate_fits(x_i, fit_key)
+                y_i, y_i_err = tmp.T[0], tmp.T[1]
+                y_f = eval_emriChif(2*self.qEmriMax, x[1:4])
+                y_f_err = 2*np.abs(x[1:4] - y_f)
+                y = (y_f - y_i) * np.sin(np.pi/(2*self.qEmriMax)* \
+                    (x[0] - self.qEmriMax))**2 + y_i
+                y_err = (y_f_err - y_i_err) * np.sin(np.pi/(2*self.qEmriMax)* \
+                    (x[0] - self.qEmriMax))**2 + y_i_err
+            else:
+                y = eval_emriChif(x[0], x[1:4])
+                y_err = 2*np.abs(x[1:4] - y)
+
+            return y, y_err
+
+        def eval_vector_fit(x, fit_key):
+            if fit_key == 'chif':
+                if x[0] > self.qEmriMax:
+                    fit_val, fit_err = extrap_chif(x, fit_key)
+                else:
+                    res = self._evaluate_fits(x, fit_key)
+                    fit_val = res.T[0]
+                    fit_err = res.T[1]
+            
+            return fit_val, fit_err
+        
+        """
+        The remnant mass fit is enforced through a mapping procedure in 
+        order to include physical constraint (mf/M < 1, M = m1+m2). The fit is 
+        trained passing arctanh(mf) and it returs the tanh, so the values are
+        constrained in the interval [0,1). See Sec. III in THE PAPER for 
+        additional details.
+        """
+        if fit_key == 'mf' or fit_key == 'all':
+            mf, mf_err = self._evaluate_fits(x, 'mf')
+            mf, mf_err = np.tanh(mf), 1/np.cosh(mf)**2*mf_err
```

Comment:
Just so I understand, does the `1/np.cosh(mf)` come from doing the error propagation properly?

Reply:
Yes